library/src/main/java/com/fraggjkee/smsconfirmationview/SmsConfirmationView.kt

```diff
@@ -100,6 +112,55 @@ class SmsConfirmationView @JvmOverloads constructor(
         }
     }
 
+    // Used to inflate and return the popup paste menu
+    private fun initPasteMenu() : PopupMenu {
+        val clipboard = getSystemService(context, ClipboardManager::class.java)
+        val pasteMenu = PopupMenu(this.context, this).apply {
+            inflate(R.menu.popup_menu_paste)
+            setOnMenuItemClickListener { menuItem ->
+                when (menuItem.itemId) {
+                    R.id.menu_item_paste -> {
+                        // Examines the item on the clipboard. If getText() doesn't return null,
+                        // the clip item contains the text. Assumes that this application can only
+                        // handle one item at a time.
+                        val item = clipboard?.primaryClip?.getItemAt(0)
+
+                        // Get the clipboard item text.
+                        val pasteData = item?.text?.toString() ?: ""
+
+                        // Paste the text into the component
+                        appendPaste(pasteData)
```

Comment:
we should add some filtering for the pasted content: it is now possible to paste text and not just digits. I guess we can just extend `appendSymbol` a little bit to skip non-digit symbols